### PR TITLE
require Node >= 4

### PR DESCRIPTION
**Pull request by stephenplusplus**
This PR changes our implicit dependency on Node * to explicitly depending on Node >= 4. It will require a semver-major bump to 0.7.0.

In this PR:

- Bump Node support to >= 4
- Remove `object-assign` dependency and use Object.assign instead
- Update dependencies
- Small lint fixes

@SimenB would you mind taking a look to make sure I didn't gloss over anything?

### CODE REVIEW

**Comment by stephenplusplus** [comment]
Done! Would you take another look at my arrow function-ization and classifying?

File: index.js [hunk 1]
```diff
@@ -1,241 +1,232 @@
...
-    return;
   }
+}
 
-  gcpMetadata.instance('/attributes/cluster-name', function (err) {
-    self.environment.IS_CONTAINER_ENGINE = !err;
-
-    callback(null, self.environment.IS_CONTAINER_ENGINE);
-  });
+module.exports = config => {
```
Comment (stephenplusplus) [inline index.js]: I'm not very experienced with some of the new ES features (like Classes). Is there a way to support lazy instantiation, e.g.:

```js
var googleAutoAuth = require('google-auto-auth')

var authClient = new googleAutoAuth()
// and
var authClient = googleAutoAuth()
```

Comment (SimenB) [inline index.js]: I have no experience with JS classes, so can't really comment on that

Comment (stephenplusplus) [inline index.js]: @callmehiphop any idea on this?

**Pull request merged**